src/parseISODuration/index.ts

```diff
@@ -0,0 +1,73 @@
+import type { Duration } from "../types.js";
+
+const ISO_DURATION_REGEX = /P(?:([\d]+\.?[\d]*|\.[\d]+)Y)?(?:([\d]+\.?[\d]*|\.[\d]+)M)?(?:([\d]+\.?[\d]*|\.[\d]+)W)?(?:([\d]+\.?[\d]*|\.[\d]+)D)?(?:T(?:([\d]+\.?[\d]*|\.[\d]+)H)?(?:([\d]+\.?[\d]*|\.[\d]+)M)?(?:([\d]+\.?[\d]*|\.[\d]+)S)?)?$/;
+
+/**
+ * @name parseISODuration
+ * @category Common Helpers
+ * @summary Parse ISO Duration string
+ *
+ * @description
+ * Parse the given string in ISO 8601 format and return Duration object.
+ *
+ * Function accepts duration format of ISO 8601.
+ * ISO 8601: http://en.wikipedia.org/wiki/ISO_8601
+ * Specifically: https://en.wikipedia.org/wiki/ISO_8601#Durations
+ *
+ * If the argument is not a valid ISO Duration it will throw Invalid format error.
+ *
+ * @param argument - ISO Duration string
+ * @returns Parsed duration object
+ *
+ * @throws It throws an "Invalid format" error if the string doesn't match the ISO Duration format
+ *
+ * @example
+ * // Convert string 'P3Y6M4DT12H30M5S' to duration:
+ * const result = parseISODuration('P3Y6M4DT12H30M5S')
+ * //=>
+ * const result = {
+ *   years: 3,
+ *   months: 6,
+ *   weeks: 0,
+ *   days: 4,
+ *   hours: 12,
+ *   minutes: 30,
+ *   seconds: 5,
+ * }
```

Comment:
@matinzd I think what we need here is for the entire result object to be formatted as a comment. You can refer to `src/intervalToDuration/index.ts` for an example.

```suggestion
 * //=> { years: 3, months: 6, weeks: 0, days: 4, hours: 12, minutes: 30, seconds: 5 }
```